app/topics/topic-management/add-existing-subtopic/add-existing-subtopic.component.ts

```diff
@@ -36,12 +34,24 @@ export class AddExistingSubtopicComponent {
     this.showChange.emit(this.show);
   }
 
+  private filterTopics() {
+    for(var subtopic of this.subtopics) {
+      for(var searchResult of this.searchResults) {
```

Comment:
I tried using indexOf for searchResults array, but it was not working; it was always returning -1, so I splice whole searchResult which was working correctly.